src/python/pants/help/help_formatter.py

```diff
@@ -38,6 +38,7 @@ def add_option(ohis, *, category=None):
                 lines.append(self.maybe_green(f"{title}\n{'-' * len(title)}\n"))
                 lines.extend(hard_wrap(oshi.description, width=self._width))
                 lines.append(" ")
+                lines.append(f"Provider: {self.maybe_magenta(oshi.provider)}")
```

Comment:
Using `provider` here, as it may come from either a `backend` or a `plugin`...
Could of course have a toggle, if the provider begins with `pants.` it's a backend, otherwise a plugin...